pandera/backends/polars/components.py

```diff
@@ -85,7 +239,7 @@ def check_dtype(
 
         if schema.dtype is not None:
             obj_dtype = check_obj.schema[schema.name]
-            passed = obj_dtype.is_(schema.dtype)
+            passed = obj_dtype.is_(schema.dtype.type)
```

Comment:
yes, I'm planning on working on that in a separate PR. It does require support in `PolarsData(dataframe: pl.LazyFrame, key: str)` as an input, since we want to support LazyFrames.